x-pack/plugins/session_view/public/components/process_tree/hooks.ts

```diff
@@ -250,5 +263,13 @@ export const useProcessTree = ({ sessionEntityId, data, searchQuery }: UseProces
 
   sessionLeader.orphans = orphans;
 
+  // update alert status in processMap for alerts in updatedAlertsStatus
+  Object.keys(updatedAlertsStatus).forEach((alertUuid) => {
+    const process = processMap[updatedAlertsStatus[alertUuid].processEntityId];
+    if (process) {
+      updateAlertEventStatus(process.getAlerts(), updatedAlertsStatus);
```

Comment:
since `getAlerts()` comes from `ProcessImpl`, what about adding an update function to `ProcessImpl`? i.e `process.setAlerts()`, to make more explicit the intention of changing alerts in the process

then also, could change to a more functional approach using map transforms:

i.e:

```js
process.setAlerts(updateAlertEventStatus(process.getAlerts(), updatedAlertsStatus))
```

```js
// updateAlertEventStatus

export const updateAlertEventStatus = (
  events: ProcessEvent[],
  updatedAlertsStatus: UpdateAlertStatus
) => (
events.map((event) => ({
    ...event, kibana: { ...event.kibana, alert: {
        ...event.kibana.alert,
        workflow_status : updatedAlertsStatus[event.kibana.alert.uuid].status
   }}
)})
```

Reply:
The alerts currently are placed together with events in `process.events`. It's probably a good idea to separate them to `process.events` and `process.alerts`. It seems to relate to some changes made in https://github.com/elastic/kibana/pull/127500 in efforts to separate fetching events and alerts. Should be make the optimization in a separate PR?

cc: @mitodrummer

Reply:
Yea, separate PR seems best. Maybe after i merge the alerts tab stuff.

Reply:
yeah another PR sounds nice to me too